ruby_basics/lib/election_winner.rb

```diff
@@ -0,0 +1,13 @@
+# Implement a function which calculates the winner of an election. Each person's vote is represented
+# as a single candidate name. The candidate with the highest number of votes wins the election. In
+# case of a tie, when multiple candidates have the same number of votes, they should be sorted
+# alphabetically in descending order where last name wins. If Albert and Zach both get 5 votes then
+# the winner is Zach.
+#
+# Eg.
+# if votes are ['Paula', 'Zach', 'Albert', 'Zach', 'Paula', 'Albert', 'Albert', 'Zach']
+# winner is 'Zach'
+
+def election_winner(votes)
+  votes.tally.to_a.max_by { |name, vote| [vote, name] }.first
```

Comment:
Isto kao i gore, ovaj `to_a` je nepotreban. Jedini efekt je da se konstruira jedna medju array koji se onda odmah odbaci.

Vecina ovih metoda je definirana na `Enumerable`. U Ruby programiranju je jako bitno da ti sjedne sta je tocno `Enumerable` tako da ako imas ikakve nedoumice, to je bitno da prodjemo u zivo.